css/css-transitions-1/events-004.html

```diff
@@ -1,150 +1,61 @@
-                // the global suite timeout
-                timeout: timeout,
-                // prepare individual test
-                setup: function(data, options) {
-                    var styles = {
-                        '.fixture': {},
-
-                        '.container': {},
-                        '.container.to': {},
-                        '.container.how': {},
-
-                        '.transition': data.from,
-                        '.transition.to' : data.to,
-                        '.transition.how' : {}
-                    };
-
-                    generalParallelTest.setup(data, options);
-                    generalParallelTest.addStyles(data, options, styles);
-                },
-                // cleanup after individual test
-                teardown: generalParallelTest.teardown,
-                // invoked prior to running a slice of tests
-                sliceStart: generalParallelTest.sliceStart,
-                // invoked after running a slice of tests
-                sliceDone: generalParallelTest.sliceDone,
-                // test cases, make them as granular as possible
-                cases: {
-                    // test TransitionEnd events
-                    'events': {
-                        start: function(test, data, options) {
-                            // kick off the transition
-                            generalParallelTest.startTransition(data);
-                        },
-                        done: function(test, data, options) {
-                            // make sure we got the event for the tested property only
-                            test.step(generalParallelTest.assertExpectedEventsFunc(data, 'transition', data.expect));
-                        }
-                    }
-                },
-                // called once all tests are done
-                done: generalParallelTest.done
-            });
-        </script>
-    </body>
+<head>
+<meta charset=utf-8>
+<title>CSS Transitions Test: transitionend event with non matching lists</title>
+<meta name="assert" content="Test checks that non-matching lists are properly resolved">
+<link rel="help" title="2. Transitions - Example 3" href="http://www.w3.org/TR/css3-transitions/#list-matching">
+<link rel="help" title="5. Transition Events" href="http://www.w3.org/TR/css3-transitions/#transition-events">
+<link rel="author" title="Rodney Rehm" href="http://rodneyrehm.de/en/">
+
+<script src="/resources/testharness.js" type="text/javascript"></script>
+<script src="/resources/testharnessreport.js" type="text/javascript"></script>
+<script src="./support/helper.js" type="text/javascript"></script>
+
+</head>
+<body>
+
+<div id="log"></div>
+
+<script>
+promise_test(t => {
+  const div = addDiv(t, { style: 'padding: 1px' });
+  getComputedStyle(div).paddingLeft;
+
+  div.style.transitionProperty =
+    'padding-top, padding-right, padding-bottom, padding-left';
+  div.style.transitionDuration = '0.02s, 0.01s';
```

Comment:
Should the durations be left unchanged for this test?

Reply:
They probably could in this case. In general I've adjusted the durations to be as short as possible while still being significantly more than the tolerance we allow when comparing times so that when we run them in automation that don't add up to a significant amount of machine time.

So in this case I've changed `0.1s, 0.05s` (total 150ms) to `0.02s, 0.01s` (30ms) which is not a massive saving but between ~10 test cases means >1s saving. Some of the other tests are more significant.